### ISSUE TITLE
Doesn't work on gnome 3.18

### ISSUE THREAD

**Issue by JodiTheTigger** [wilfm/GnomeExtensionMaximusTwo#42]
ARch 64, just upgraded to gnome 3.18, extension didn't work.

I updated metadata.json from 3.16 to 3.18 and tried again, I got an error in lg of:

> No signal 'maximize' on object 'ShellWM'

**Comment by JodiTheTigger** [comment]
Other people with the same issue:
https://github.com/mlutfy/hidetopbar/issues/79
https://github.com/micheleg/dash-to-dock/issues/214
https://github.com/micheleg/dash-to-dock/issues/202

**Comment by JodiTheTigger** [comment]
As per (https://github.com/mlutfy/hidetopbar/pull/82/files) I changed

```
    wmCallbackIDs.push(global.window_manager.connect('maximize', updateAppMenu));
    wmCallbackIDs.push(global.window_manager.connect('unmaximize', updateAppMenu));
```

in app_menu.js (~216) to

```
    wmCallbackIDs.push(global.window_manager.connect('size-change', updateAppMenu));
```

So now the script runs, but doesn't remove the title bar.

I give up :-(

**Comment by davidcortesortuno** [comment]
Same here

**Comment by viing937** [comment]
I do as you said and it works well now.
Arch 64, too.

**Comment by jonrouleau** [comment]
@JodiTheTigger 
@davidcortesortuno 

> So now the script runs, but doesn't remove the title bar.

Make sure you have `xprop` installed

On Arch: `pacman -S xorg-utils`

See: https://github.com/wilfm/GnomeExtensionMaximusTwo#prerequisites

**Comment by JodiTheTigger** [comment]
@jrouleau Thanks, didn't notice the dependencies, dunno why it worked previously :-/

**Comment by wilfm** [comment]
Any issues in 3.18 with `xprop` etc installed?

**Comment by ikuraj** [comment]
Not sure if this is related, but I used the extension with 3.18 (installed through this [aur package](https://aur.archlinux.org/packages/gnome-shell-extension-maximus-two-git/)) and noticed constant CPU usage (on my machine it amounts to 20%) and some errors in the log like:

```
Nov 04 20:40:17 pc gnome-session[2092]: Gjs-Message: JS LOG: [maximus-two]: Can't find original state for null with id 0x40016b8
Nov 04 20:40:41 pc gnome-session[2092]: Gjs-Message: JS LOG: [maximus-two]: Can't find original state for Unsaved Document 1 - gedit with id 0x6600059
```

**Comment by wilfm** [comment]
@kaptoxic thats not from this repo, thats from someone else's (this one https://github.com/dffischer/GnomeExtensionMaximusTwo), so you could report it there. You could try the https://github.com/danielkza/GnomeExtensionMaximusTwo/ branch though (currently for just 3.16), or the one from https://github.com/wilfm/GnomeExtensionMaximusTwo/pull/43.

I know I need to merge and update stuff myself, its just A) merging the right, adding bits etc B) testing (difficult at the mo) C) getting past the reviewing thing (unlike version 4 :( ). Sorry...

**Comment by ikuraj** [comment]
Oh, I see. I thought that AUR package was based on your repo (I guess it got changed). Wow, these (evidently popular) forks are two-level forks of your repo. :)

Yes, this seems a mess. How hard is to consolidate these repositories into one? How can the community help you in doing this (especially the testing part)?

**Comment by hAh0L13** [comment]
Can't install in gnome 3.18 under x64 Fedora 23.

Try to install from extensions.gnome.org - switch to "On", agree on pop-up window, but extension doesn't install. When i refresh page, switch seems to be "Off"

Try to git clone, zip and install from gnome-tweak-tool, but it says "Invalid extension"

PS Previously used pixel-saver, first disable it before install MaximusTwo.

**Comment by diaevd** [comment]
Temporary you can clone extension from https://github.com/diaevd/GnomeExtensionMaximusTwo
Just try it.

**Comment by hAh0L13** [comment]
Invalid extension, as i previously say.

**Comment by diaevd** [comment]
You need to zip only maximus-two@wilfinitlike.gmail.com folder from cloned repo, but not the whole folder.
Or just copy maximus-two@wilfinitlike.gmail.com to ~/.local/share/gnome-shell/extensions  and then enable it:

git clone https://github.com/diaevd/GnomeExtensionMaximusTwo
cd GnomeExtentionMaximusTwo
cp -R maximus-two@wilfinitlike.gmail.com  ~/.local/share/gnome-shell/extensions/
gnome-shell-extension-tool -e "maximus-two@wilfinitlike.gmail.com"

**Comment by hAh0L13** [comment]
Yeah, i'm try to zip only maximus-two folder, try to zip only files in that folder - anyway, "Invalid extension"

Second option - copy to .local and use commmand-line tool help me. It says "already enabled" and after shell restart MaximusTwo successfully work for me, big thanks!

PS I don't know what i did wrong

**Comment by Svarozic** [comment]
I have also Fedora23 with Gnome 3.18 but for my this plugin works out of box: https://github.com/danielkza/GnomeExtensionMaximusTwo/

I have just downloaded it and extracted "maximus-two@wilfinitlike.gmail.com" to extensions folder ..

**Comment by johnche** [comment]
Im running Fedora23 and are also having troubles.

Seems that its being labeled as outdated in my gnome-extensions. I did copy "maximus-two@wilfinitlike.gmail.com" to extensions folder and ran gnome-shell-extension-tool -e on it without success
![screenshot from 2016-04-07 21-58-29](https://cloud.githubusercontent.com/assets/10786387/14365068/096fbedc-fd0c-11e5-8ba5-60098ae9dd10.png)

Edit:
I figured the solution to my problem was that metadata.json in maximus-two@wilfinitlike.gmail.com folder has a 3.16 that should be changed to 3.18 so to fit with my gnome shell version.

**Comment by ikuraj** [comment]
@wilfm, is there any chance you can describe shortly (e.g. in a wiki) how does this extension work and where should we look at if we want to make it work, please?
(Your extension is a lifesaver, especially since titlebars in Gnome 3.20 became even bigger for some reason and it's really annoying.)

**Comment by ikuraj** [comment]
Just to confirm that I was running the version from the master branch of [this repo](https://github.com/danielkza/GnomeExtensionMaximusTwo) for several days and everything seems to be working fine -- no high CPU usage or other visible issues (on Arch Linux, Gnome 3.20.2).

Therefore, I believe all these different versions/branches could be removed and/or consolidated, and these reported issues (finally) closed.

**Comment by tongtf** [comment]
I change the file 
/home/user/.local/share/gnome-shell/extensions/maximus-two@wilfinitlike.gmail.com/metadata.json
item "shell-version"    from 3.16  to 3.20 ,it work well under fedora 24,gnome 3.20
